app/src/features/InntektsmeldingSkjemaState.tsx

```diff
@@ -0,0 +1,54 @@
+import type { ReactNode } from "@tanstack/react-router";
+import type { Dispatch, SetStateAction } from "react";
+import { createContext, useContext, useState } from "react";
+
+export type InntektsmeldingSkjemaState = {
+  kontaktperson?: { navn: string; telefon: string };
+  korrigertMånedslønn?: number;
+  skalRefunderes?: boolean;
+  refusjonsbeløpPerMåned?: number;
+  endringIRefusjon?: boolean;
+  refusjonsendringer?: { måned: string; beløp: number }[];
+  misterNaturalytelser?: boolean;
+  naturalytelserSomMistes?: { navn: string; beløp: number; fraOgMed: string }[];
+};
+
+type InntektsmeldingSkjemaStateContextType = {
+  inntektsmeldingSkjemaState: InntektsmeldingSkjemaState;
+  setInntektsmeldingSkjemaState: Dispatch<
+    SetStateAction<InntektsmeldingSkjemaState>
+  >;
+};
+const InntektsmeldingSkjemaStateContext =
+  createContext<InntektsmeldingSkjemaStateContextType | null>(null);
+
+type InntektsmeldingSkjemaStateProviderProps = {
+  children: ReactNode;
+};
+export const InntektsmeldingSkjemaStateProvider = ({
+  children,
+}: InntektsmeldingSkjemaStateProviderProps) => {
+  const [state, setState] = useState<InntektsmeldingSkjemaState>({});
+  return (
+    <InntektsmeldingSkjemaStateContext.Provider
+      value={{
+        inntektsmeldingSkjemaState: state,
+        setInntektsmeldingSkjemaState: setState,
+      }}
```

Comment:
Dette er vel _egentlig_ en perf faux pas, men tipper det går fint. Får sjekke litt når vi er i mål.

Reply:
Må innrømme jeg ikke kjenner til den terminologien så vet ikke hva du prøver si her :P 

Kan godt bruke context, men siden vi bruker tanstack router er jeg litt fan av å bruke search params som [OG state manager](https://tanstack.com/router/latest/docs/framework/react/guide/search-params#search-params-the-og-state-manager).
Har et par fordeler ved at man ikke mister all state om man ender med å navigere i nettleseren.